### PR TITLE
Use mapbox-gl-js-supported

### LINKED ISSUE #2425: Move mapboxgl.supported logic into standalone module
Presently, end users are required to include all of `mapbox-gl.js` in order to test if it is supported. This is impractical for bandwidth-conscious users who are implementing fallback to `mapbox.js`.

### PR DESCRIPTION
**Pull request by lucaswoj**
cc @ansis @mourner @bhousel @jfirebaugh @danswick 

fixes #2425

### CODE REVIEW

File: js/util/browser/canvas.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ Canvas.prototype.resize = function(width, height) {
     this.canvas.style.height = height + 'px';
 };
 
+// If you update this object, you must also update mapbox-gl-js-supported
+// https://github.com/mapbox/mapbox-gl-js-supported/blob/gh-pages/index.js#L94
```
Comment (bhousel) [inline js/util/browser/canvas.js]: Why not just export them from the other module?

Comment (lucaswoj) [inline js/util/browser/canvas.js]: I felt funny about having `mapbox-gl-js-supported` "own" this data. 

I agree that moving it to`mapbox-gl-js-supported`  carries rather significant advantage. I will adjust my approach.

Comment (lucaswoj) [inline js/util/browser/canvas.js]: Fixed in the latest commit

**Comment by lucaswoj** [comment]
@bhousel: Is this ready to 🚢?

**Comment by bhousel** [comment]
LGTM 👍

**Comment by davidtheclark** [comment]
Is `mapboxgl.supported` going to be removed? It looks to me like the new module `mapbox-gl-js-supported` is still being built into `mapbox-gl-js`: https://github.com/mapbox/mapbox-gl-js/blob/master/js/util/browser/browser.js#L77. And in studio that property is being used, I believe: https://github.com/mapbox/studio/blob/51303716353594adbafc1c3c6d0507f6c98948f8/src/util/supported.js#L6

cf. #2709

**Comment by lucaswoj** [comment]
@davidtheclark We plan to continue embedding `mapbox-gl-js-supported` within `mapbox-gl-js` as well as making it available as an external module.

**Comment by davidtheclark** [comment]
@lucaswoj Thanks for the clarification. Should I submit a PR adding it to the documentation page, or is there a reason it's not there?

**Comment by lucaswoj** [comment]
@davidtheclark I'm not sure why this isn't documented. A PR would be great! Thank you. 🙇

**Pull request merged**